cmd/quickdata/main.go

```diff
@@ -0,0 +1,468 @@
+		return quickdata.OrderBookFocusType, nil
+	case "kline", "candles", "candle", "ohlc":
+		return quickdata.KlineFocusType, nil
+	case "trades", "trade":
+		return quickdata.TradesFocusType, nil
+	case "openinterest", "oi":
+		return quickdata.OpenInterestFocusType, nil
+	case "fundingrate", "funding":
+		return quickdata.FundingRateFocusType, nil
+	case "accountholdings", "account", "holdings", "balances":
+		return quickdata.AccountHoldingsFocusType, nil
+	case "activeorders", "orders":
+		return quickdata.ActiveOrdersFocusType, nil
+	case "orderexecution", "executionlimits", "limits":
+		return quickdata.OrderLimitsFocusType, nil
+	case "url", "tradeurl", "trade_url":
+		return quickdata.URLFocusType, nil
+	case "contract":
+		return quickdata.ContractFocusType, nil
+	default:
+		return quickdata.UnsetFocusType, quickdata.ErrUnsupportedFocusType
+	}
+}
+
+func fatalErr(err error) {
+	_, _ = fmt.Fprintln(os.Stderr, err)
+	os.Exit(1)
+}
+
+// -------- Rendering helpers --------
+
+const (
+	ansiClear = "\033[2J\033[H"
+	ansiReset = "\033[0m"
+	ansiGreen = "\033[32m"
+	ansiRed   = "\033[31m"
+	ansiDim   = "\033[2m"
+	ansiBold  = "\033[1m"
+)
+
+func clearScreen() {
+	outPrint(ansiClear)
+}
+
+func renderOrderbook(b *orderbook.Book, levels int) {
+	if levels <= 0 {
+		levels = 15
+	}
+	outPrintf("%s%-14s %-14s %-14s%s\n", ansiDim, "Price", "Amount", "Total", ansiReset)
+	// Asks (red), display from best (lowest) up to levels
+	askCount := intMin(levels, len(b.Asks))
+	cumulativeAsks := 0.0
+	for i := range askCount {
+		lvl := b.Asks[i]
+		cumulativeAsks += lvl.Amount
+		outPrintf("%s% -14.8f % -14.8f % -14.8f%s\n", ansiRed, lvl.Price, lvl.Amount, cumulativeAsks, ansiReset)
+	}
+	outPrintln()
+	// Bids (green), from best (highest) up to levels
+	bidCount := intMin(levels, len(b.Bids))
+	cumulativeBids := 0.0
+	for i := range bidCount {
+		lvl := b.Bids[i]
+		cumulativeBids += lvl.Amount
+		outPrintf("%s% -14.8f % -14.8f % -14.8f%s\n", ansiGreen, lvl.Price, lvl.Amount, cumulativeBids, ansiReset)
+	}
+}
+
+func renderTicker(t *ticker.Price) {
+	if t == nil {
+		return
+	}
+	spread := t.Ask - t.Bid
+	var spreadPct float64
+	if t.Last != 0 {
+		spreadPct = (spread / t.Last) * 100
+	}
+	outPrintf("%sLast:%s %.8f  %sBid:%s %.8f  %sAsk:%s %.8f\n",
+		ansiBold, ansiReset, t.Last, ansiGreen, ansiReset, t.Bid, ansiRed, ansiReset, t.Ask)
+	outPrintf("%sVol:%s  %.4f  %sSpread:%s %.8f (%.4f%%)  %sMark:%s %.8f  %sIndex:%s %.8f\n",
+		ansiDim, ansiReset, t.Volume, ansiDim, ansiReset, spread, spreadPct, ansiDim, ansiReset, t.MarkPrice, ansiDim, ansiReset, t.IndexPrice)
+}
+
+func renderTrades(trs []trade.Data) {
+	if len(trs) == 0 {
+		outPrintln("No trades.")
+		return
+	}
+	// Summaries
+	n := len(trs)
+	start := trs[0].Timestamp
+	end := trs[n-1].Timestamp
+	if end.Before(start) {
+		start, end = end, start
+	}
+	var baseVol, quoteVol float64
+	var buys, sells int
+	for i := range trs {
+		baseVol += trs[i].Amount
+		quoteVol += trs[i].Amount * trs[i].Price
```

Comment:
Shouldn't rename this `.Amount` variable to something like `.Quantity`? because most of the time, it is used to express the quote amount and here it is the base amount.

Reply:
That is part of the trade package and I'll say that's out of scope for this part here